### ISSUE TITLE
Define directory structure for source and bytecode files

### ISSUE THREAD

**Issue by yorickpeterse** [inko-lang/inko#194]
The compiler currently uses ~/.cache/inko as a default target directory for bytecode. Projects in turn will use a structure where their source code is in a `src/` directory, and tests in `tests/`. However, there's no specification about where things go when you install a package, or when you were to install Inko the proper way (e.g. via a package manager). For example, installing Inko via a package manager might lead to the following configuration:

1. Base library directory: `/usr/lib/inko/VERSION/`
1. Standard library: `/usr/lib/inko/VERSION/runtime/std`
1. Core library: `/usr/lib/inko/VERSION/runtime/core`
1. Bytecode output: `$XDG_CACHE_HOME/inko/target/TARGET`, with TARGET being `release`, `debug`, `test`, etc.
1. Third-party source code: `/usr/lib/inko/VERSION/pkg`. Version 1.0.0. of package `foo` would then be located in `/usr/lib/inko/VERSION/pkg/foo/1.0.0/...`

For user local installations this would instead be something like:

1. Base library directory: `~/.local/share/inko/VERSION/`
1. Standard library: `~/.local/share/inko/VERSION/runtime/std`
1. Core library: `~/.local/share/inko/VERSION/runtime/core`
1. Bytecode output: `$XDG_CACHE_HOME/inko/target/TARGET`
1. Third-party source code: `~/.local/share/inko/VERSION/pkg`

In other words: there are two prefixes for all directories:

1. A prefix for all source files, manuals, binaries, etc. Let's call the variable that controls this `INKO_HOME`.
2. A prefix for any temporary files, including bytecode. The variable that controls this is `INKO_CACHE`

Based on the above, the defaults would be:

1. `INKO_HOME`: `/usr/lib/inko`
1. `INKO_CACHE`: `$XDG_CACHE_HOME/inko`, which usually translates to `~/.cache/inko`

All of this should be turned into some kind of specification.
